exec-service-wasmer/src/wasmer_executor.rs

```diff
@@ -1,101 +1,62 @@
-        match value {
-            0 => Ok(WasmerExecutorLogLevel::None),
-            1 => Ok(WasmerExecutorLogLevel::Debug),
-            2 => Ok(WasmerExecutorLogLevel::Trace),
-            _ => Err("WasmerExecutor undefined log level"),
-        }
-    }
-}
+use std::sync::{Arc, Mutex};
 
 pub struct WasmerExecutorData {
     pub vm_hooks: Rc<Box<dyn VMHooks>>,
-    pub opcode_cost: Arc<OpcodeCost>,
-    pub log_level: WasmerExecutorLogLevel,
+    pub opcode_cost: Arc<Mutex<OpcodeCost>>,
 }
 
 impl WasmerExecutorData {
-    pub(crate) fn debug(&self, message: &str) {
-        if self.log_level >= WasmerExecutorLogLevel::Debug {
-            println!("[DEBUG] {message}");
+    fn set_vm_hooks_ptr(&mut self, vm_hooks_ptr: *mut c_void) -> Result<(), ExecutorError> {
+        if let Some(vm_hooks) = Rc::get_mut(&mut self.vm_hooks) {
+            vm_hooks.set_vm_hooks_ptr(vm_hooks_ptr);
+            return Ok(());
         }
+
+        Err(Box::new(ServiceError::new(
+            "WasmerExecutor already set vmhooks, further configuration not allowed",
+        )))
     }
-    #[allow(dead_code)]
-    pub(crate) fn trace(&self, message: &str) {
-        if self.log_level == WasmerExecutorLogLevel::Trace {
-            println!("[TRACE] {message}");
-        }
+
+    fn set_opcode_cost(&mut self, opcode_cost: &OpcodeCost) -> Result<(), ExecutorError> {
+        self.opcode_cost.lock().unwrap().clone_from(opcode_cost);
+        Ok(())
     }
 }
 
+pub struct WasmerExecutor {
+    pub data: Rc<RefCell<WasmerExecutorData>>,
+}
+
 impl Executor for WasmerExecutor {
     fn set_vm_hooks_ptr(&mut self, vm_hooks_ptr: *mut c_void) -> Result<(), ExecutorError> {
-        self.data.debug("Setting context pointer ...");
-        if let Some(data_mut) = Rc::get_mut(&mut self.data) {
-            if let Some(vm_hooks) = Rc::get_mut(&mut data_mut.vm_hooks) {
-                vm_hooks.set_vm_hooks_ptr(vm_hooks_ptr);
-                return Ok(());
-            }
-        }
-
-        Err(Box::new(ServiceError::new(
-            "WasmerExecutor already has instances, can no longer be configured",
-        )))
+        info!("Setting vmhooks ...");
+        self.data.borrow_mut().set_vm_hooks_ptr(vm_hooks_ptr)
     }
 
     fn set_opcode_cost(&mut self, opcode_cost: &OpcodeCost) -> Result<(), ExecutorError> {
-        self.data.debug("Setting opcode cost ...");
-        if let Some(data_mut) = Rc::get_mut(&mut self.data) {
-            if let Some(opcode_cost_mut) = Arc::get_mut(&mut data_mut.opcode_cost) {
-                *opcode_cost_mut = opcode_cost.clone();
-                return Ok(());
-            }
-        }
-
-        Err(Box::new(ServiceError::new(
-            "WasmerExecutor opcodes cost configuration error",
-        )))
+        info!("Setting opcode cost...");
+        self.data.borrow_mut().set_opcode_cost(opcode_cost)
     }
 
     fn set_execution_log_level(&mut self, value: u64) -> Result<(), ExecutorError> {
```

Comment:
We should move this to the service, or CAPI

Reply:
Right, already noted this down in description as todo

Reply:
Done